firmware/quadruna/VC/src/app/vehicle_dynamics/app_tractionControl.c

```diff
@@ -21,18 +21,31 @@ void app_tractionControl_computeTorque(TractionControl_Inputs *inputs, TractionC
 
     float slip_ratio_max = fmaxf(slip_ratio_left, slip_ratio_right);
     float k              = app_pid_compute(pid, SLIP_RATIO_IDEAL, slip_ratio_max);
+    static float correction_torque; 
+
+    // NOTE: we want to map K to a vlaue that we can multiple with input torque. K is based on the slip error 
+    // we want to map it to torque error, we also want to  make sure we use the torque associated to the wheel from which slip max was taken
+
+    if( slip_ratio_max == slip_ratio_left){
```

Comment:
are both wheels gonna have uniform conditions? probably not right so why would u apply correction torque to both? also where did u get this correcttion torque equation from that uses wheel speed/ vehicle speed. Could do it like this ?
```
if (slip_ratio_max == slip_ratio_left) {
correction_torque = k * inputs->torque_left_Nm;    
outputs->torque_left_final_Nm  = inputs->torque_left_Nm + correction_torque;
    outputs->torque_right_final_Nm = inputs->torque_right_Nm;  
} else {
    correction_torque = k * inputs->torque_right_Nm; 
outputs->torque_right_final_Nm = inputs->torque_right_Nm + correction_torque;
    outputs->torque_left_final_Nm  = inputs->torque_left_Nm; 
}
```

Reply:
yeah that correction torque calculation is wrong. what I was trying to do is map it make to a torque value but I did it wrong... in the future it would be good to map it to a vehicle speed using vehicle speed = (1 +- slip) (wheel_speed) depending on acceleration, then turn that into an angular velocity and then place it in a queue of sleep values from which we can take two entries from to have like a derivative to get acceleration then convert that into a torque with T = M * and_a 


I will change it to what you said